Sources/MongoSwift/ReadConcern.swift

```diff
@@ -86,13 +115,19 @@ public class ReadConcern: Codable {
         try container.encodeIfPresent(self.level, forKey: .level)
     }
 
-    /// Cleans up internal state.
-    deinit {
-        guard let readConcern = self._readConcern else {
-            return
+    /**
+     * Creates a new `ReadConcern` with the provided options and passes it to the provided closure.
+     * The read concern is only valid within the body of the closure and will be ended after the body completes.
+     */
+    public func withMongocReadConcern<T>(_ body: (OpaquePointer) throws -> T) throws -> T {
```

Comment:
So it appears this function only throws an error if the closure passed to it (`body`) also throws. None of the work you do here to create the `mongoc_read_concern_t` throws, you just propagate the error from `body`.

for that reason, we should mark this function as `rethrows` rather than `throws`. here's a nice little blog post on [`rethrows`](https://www.hackingwithswift.com/example-code/language/how-to-use-the-rethrows-keyword).

this should mean that you don't actually need the `try?` / `try` at all where you are using this helper to call `mongoc_client_set_read_concern` etc.